config/checkstyle-checks.xml

```diff
@@ -581,6 +581,14 @@
     <module name="UnnecessarySemicolonAfterTypeMemberDeclaration"/>
     <module name="UnnecessarySemicolonInEnumeration"/>
     <module name="UnnecessarySemicolonInTryWithResources"/>
+    <module name="UnusedCatchParameterShouldBeUnnamed">
+      <!--
+        We should not use this check until we migrate to
+        java 21. Unnamed catch parameters will not compile on older
+        java versions.
+      -->
```

Comment:
We will not use it, because we are very explicit in handling of exceptions, never ignored.
If it ever going to happen, I would like `ignore` as name, to be very very explicit and done on ou purpose .

Reply:
actually, we have a bunch of violations in our codebase, ideally, we should solve all of these and declare the catch parameter as unnamed (`_`) but we can't use this syntax now. I don't understand what stops us from using this check in the future?

> to be very very explicit and done on ou purpose .

according to Java 21 and above, if you want to be very very explicit that it is ignored on purpose you should declare it as unnamed.

```
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\it\java\com\google\checkstyle\test\base\AbstractGoogleModuleTestSupport.java:203:16: Unused catch parameter 'ex' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\it\java\org\checkstyle\base\AbstractCheckstyleModuleTestSupport.java:38:20: Unused catch parameter 'ignore' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\LocalizedMessage.java:119:16: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\Main.java:852:24: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\PackageObjectFactory.java:314:16: Unused catch parameter 'ignore' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\PackageObjectFactory.java:362:16: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\XMLLogger.java:330:24: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\checks\NewlineAtEndOfFileCheck.java:137:16: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\checks\regexp\MultilineDetector.java:115:16: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\site\SiteUtil.java:1245:20: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\utils\CommonUtil.java:232:16: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\utils\CommonUtil.java:378:16: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\main\java\com\puppycrawl\tools\checkstyle\utils\FilterUtil.java:48:16: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\checks\coding\FinalLocalVariableCheckTest.java:258:16: Unused catch parameter 'ex' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\checks\coding\IllegalInstantiationCheckTest.java:158:16: Unused catch parameter 'ex' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\checks\coding\IllegalTypeCheckTest.java:426:16: Unused catch parameter 'ex' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\checks\coding\MatchXpathCheckTest.java:188:16: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\checks\coding\MatchXpathCheckTest.java:208:16: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\checks\coding\ModifiedControlVariableCheckTest.java:121:16: Unused catch parameter 'ex' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\checks\coding\ModifiedControlVariableCheckTest.java:129:16: Unused catch parameter 'ex' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\checks\coding\ReturnCountCheckTest.java:126:16: Unused catch parameter 'ex' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\checks\coding\ReturnCountCheckTest.java:134:16: Unused catch parameter 'ex' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\filters\SuppressionFilterTest.java:263:16: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\filters\SuppressionsLoaderTest.java:227:16: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\internal\AllChecksTest.java:450:20: Unused catch parameter 'ex' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\internal\CommitValidationTest.java:306:16: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\internal\XdocsPagesTest.java:1355:20: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\internal\XdocsPagesTest.java:1815:24: Unused catch parameter 'ignore' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\internal\XdocsPagesTest.java:1888:20: Unused catch parameter 'ignore' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
[ERROR] [checkstyle] [ERROR] D:\CS\checkstyle\src\test\java\com\puppycrawl\tools\checkstyle\internal\utils\CheckUtil.java:286:16: Unused catch parameter 'ignored' should be unnamed. [UnusedCatchParameterShouldBeUnnamed]
```

Reply:
we also could update this check to ignore the names "ignore" and "ignored" and not be violated

Reply:
> we also could update this check to ignore the names "ignore" and "ignored" and not be violated

It is better to use some suppression for this; in Java 21+ the correct way to do this will be unnamed variables.

Reply:
please do wording like:
```
Unnamed catch parameters will not compile on
        java versions before 21. As we migrate to jdk 21 
   we will use xpath to avoid violation on names like 'ignored'. 
```

Reply:
please add xpath test to show that Check able to suppress by name.

Please add this xpath to issue description to let users know of migrational workaround or just workaround on existing pattern. So users can start to use it rightaway, and deal with old code later on.

Reply:
>ReturnCountCheckTest.java:126:16: Unused catch parameter 'ex' should be unnamed. 

this is bad, and user should decided to put `ignore`/`ignored` or `_` .

Reply:
> > ReturnCountCheckTest.java:126:16: Unused catch parameter 'ex' should be unnamed.
> 
> this is bad, and user should decided to put `ignore`/`ignored` or `_` .

Please restate this, I do not understand what this means.

Reply:
I meant we will fix this `ReturnCountCheckTest.java:126:16` to be named `ignored`. But it will be in future.

Reply:
Wording of comment should be as I suggested at https://github.com/checkstyle/checkstyle/pull/15358/files#r1691538791